File: src/test/java/hu/zsoki/advent/y2022/Day4Test.java
package hu.zsoki.advent.y2022;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

class Day4Test {

    @Test
    public void day4Part1TestExample() {
        final String path = ClassLoader.getSystemResource("y2022/day4_example.txt").getPath();
        try {
            List<String> lines = Files.readAllLines(Path.of(path));
            Assertions.assertEquals(2, Day4.countFullyContainedAssignments(lines));
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    @Test
    public void solveDay4Part1() {
        final String path = ClassLoader.getSystemResource("y2022/day4.txt").getPath();
        try {
            List<String> lines = Files.readAllLines(Path.of(path));
            System.out.println(Day4.countFullyContainedAssignments(lines));
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    @Test
    public void day4Part2TestExample() {
        final String path = ClassLoader.getSystemResource("y2022/day4_example.txt").getPath();
        try {
            List<String> lines = Files.readAllLines(Path.of(path));
            Assertions.assertEquals(4, Day4.countOverlappingAssignments(lines));
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    @Test
    public void solveDay4Part2() {
        final String path = ClassLoader.getSystemResource("y2022/day4.txt").getPath();
        try {
            List<String> lines = Files.readAllLines(Path.of(path));
            System.out.println(Day4.countOverlappingAssignments(lines));
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

}

File: src/test/kotlin/hu/zsoki/advent/y2023/Day4Test.kt
package hu.zsoki.advent.y2023

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class Day4Test {

    @Test
    fun day4_part1() {
        println(day4part1("y2023/day4.txt"))
    }

    @Test
    fun day4_part2() {
        println(day4part2("y2023/day4.txt"))
    }

    @Test
    fun day4_part1_example1() {
        assertEquals(13, day4part1("y2023/day4_part1_example1.txt"))
    }

    @Test
    fun day4_part2_example1() {
        assertEquals(30, day4part2("y2023/day4_part1_example1.txt"))
    }

}